src/pages/task/taskTabs/FileTableTab/index.tsx

```diff
@@ -0,0 +1,76 @@
+import { useState } from "react";
+import { useQuery } from "@apollo/client";
+import styled from "@emotion/styled";
+import { SearchInput } from "@leafygreen-ui/search-input";
+import { Skeleton, TableSkeleton } from "@leafygreen-ui/skeleton-loader";
+import { TablePlaceholder } from "components/Table/TablePlaceholder";
+import { size } from "constants/tokens";
+import { useToastContext } from "context/toast";
+import { TaskFilesQuery, TaskFilesQueryVariables } from "gql/generated/types";
+import { GET_TASK_FILES } from "gql/queries";
+import GroupedFilesTable from "./GroupedFilesTable";
+import { filterGroupedFiles } from "./utils";
+
+interface FilesTableTabProps {
+  taskId: string;
+  execution: number;
+}
+const FilesTableTab: React.FC<FilesTableTabProps> = ({ execution, taskId }) => {
+  const [search, setSearch] = useState("");
+  const dispatchToast = useToastContext();
+  const { data, loading } = useQuery<TaskFilesQuery, TaskFilesQueryVariables>(
+    GET_TASK_FILES,
+    {
+      variables: {
+        taskId,
+        execution,
+      },
+      onError: (err) => {
+        dispatchToast.error(`Unable to load task files: ${err}`);
+      },
+    }
+  );
+  const { taskFiles } = data?.task ?? {};
+
+  const { groupedFiles = [] } = taskFiles ?? {};
+  const filteredGroupedFiles = filterGroupedFiles(groupedFiles, search);
+
+  // We only want to show the file group name if there are multiple file groups.
+  const hasMultipleFileGroups = groupedFiles.length > 1;
+
+  return loading ? (
+    <FilesTableTabSkeleton />
+  ) : (
+    <>
+      <StyledSearchInput
+        aria-label="Search file names"
+        placeholder="Search file names"
+        onChange={(e) => setSearch(e.target.value)}
+        value={search}
+        data-cy="file-search-input"
+      />
+      {filteredGroupedFiles.length === 0 && (
+        <TablePlaceholder message="No Files found" />
+      )}
```

Comment:
I'd prefer to build this into the table (through `BaseTable` if possible):
<img width="1118" alt="Screenshot 2023-09-12 at 10 32 24 AM" src="https://github.com/evergreen-ci/spruce/assets/47064971/fc890185-175d-43e1-a2cf-968ea3dae2af">

but if it seems like too much effort, I feel like just using the current UI will be fine:
<img width="585" alt="Screenshot 2023-09-12 at 10 33 10 AM" src="https://github.com/evergreen-ci/spruce/assets/47064971/87b5fc5a-39ad-42a5-88cd-6827dbff430d">

Reply:
Good call expanded Base Table to support this!